src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -156,8 +156,50 @@ public static Date getMsOffsetToCurrentTimeInUserTimeZone(int offset, double tim
     }
 
     public static Calendar convertToUserTimeZone(Calendar time, double timeZone) {
-        time.add(Calendar.MILLISECOND, (int) (60 * 60 * 1000 * timeZone));
-        return time; // for chaining
+        int serverOffset = time.getTimeZone().getRawOffset();
+        int sessionOffset = (int) (60 * 60 * 1000 * timeZone);
+        int offset = sessionOffset - serverOffset;
+        TimeZone sessionTimeZone = getTimeZone(timeZone);
```

Comment:
@wkurniawan07 Until this line nothing happens, because this solution guarantees that if the method is called twice, then it won't change the timezone again, which would ruin the tests. 
I think this is a nice solution semantically, but if you just want to leave it like it was and adjust the tests accordingly, then I can do it. I am going to do that in another pull request so that you can decide which one do you like better.